### PR TITLE
Fix nil pointer deref in policy sorter.

### LINKED ISSUE #1217: nil pointer dereference in policy_sorter.go
Found by @matthewdupre: (only) when debug level logging is enabled, we print out the policies and their orders:
```
	if log.GetLevel() >= log.DebugLevel {
		names := make([]string, len(tierInfo.OrderedPolicies))
		for ii, kv := range tierInfo.OrderedPolicies {
			names[ii] = fmt.Sprintf("%v(%v)",
				kv.Key.Name, *kv.Value.Order)
		}
		log.Infof("Before sorting policies: %v", names)
	}
```
Unfortunately, `kv.Value.Order` can (legitimately) be nil so this code can try to dereference a nil pointer.

### PR DESCRIPTION
**Pull request by fasaxc**
Fixes #1217.

### CODE REVIEW

File: go/felix/calc/policy_sorter_test.go [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+	. "github.com/onsi/ginkgo/extensions/table"
+	. "github.com/onsi/gomega"
+	"github.com/projectcalico/libcalico-go/lib/backend/model"
+)
+
+var (
+	tenPointFive = 10.5
+)
+
+var _ = FDescribeTable("PolKV should stringify correctly",
```
Comment (neiljerram) [inline go/felix/calc/policy_sorter_test.go]: If I understand correctly, you need to remove the leading 'F' here (as it causes the 'Detected Programmatic Focus' status, which our CI thinks is a failure), and then retest.

**Comment by fasaxc** [comment]
> I think Order is only a pointer so that we can use the nil value to indicate 'default', right? That seems a bit hacky; is there a better alternative?

This is the backend datamodel, which maps onto our JSON; using a pointer is the natural way for the parser library to represent the presence/absence of an optional field.  (In the json, we represent default by omitting the order from the JSON, which is pretty natural, I think.)

> The new `func (p PolKV) String()` string is OK for this context - but more generally would we not want a stringification function for policies that included more information than just the name and order?

If you just do `fmt.Printf("%v", aPolicy)` then you actually get quite a lot more detail (it'll print all the fields) so the defaults are fairly good for most tracing.  In this case, I wanted to limit the verbosity to just the fields I cared about.

**Comment by neiljerram** [comment]
> > I think Order is only a pointer so that we can use the nil value to indicate 'default', right? That seems a bit hacky; is there a better alternative?

> This is the backend datamodel, which maps onto our JSON; using a pointer is the natural way for the parser library to represent the presence/absence of an optional field.

OK, so we're following the parser library here - fine.

> (In the json, we represent default by omitting the order from the JSON, which is pretty natural, I think.)

Definitely no issue there - my only query was with the code representation.

**Comment by neiljerram** [comment]
> > The new `func (p PolKV) String() string` is OK for this context - but more generally would we not want a stringification function for policies that included more information than just the name and order?

> If you just do `fmt.Printf("%v", aPolicy)` then you actually get quite a lot more detail (it'll print all the fields) so the defaults are fairly good for most tracing. In this case, I wanted to limit the verbosity to just the fields I cared about.

OK - thanks.

**Pull request merged**